bundles/org.openhab.binding.lgthinq/src/main/java/org/openhab/binding/lgthinq/internal/handler/LGThinQAbstractDeviceHandler.java

```diff
@@ -0,0 +1,820 @@
+                getLogger().warn(
+                        "Device's command queue reached the size limit. Probably the device is busy ou stuck. Ignoring command.");
+                getLogger().debug("Status of the commandQueue: consumer: {}, size: {}",
+                        commandExecutorQueueJob == null || commandExecutorQueueJob.isDone() ? "OFF" : "ON",
+                        commandBlockQueue.size());
+                if (logger.isTraceEnabled()) {
+                    // logging the thread dump to analise possible stuck thread.
+                    ThreadMXBean bean = ManagementFactory.getThreadMXBean();
+                    ThreadInfo[] infos = bean.dumpAllThreads(true, true);
+                    String message = "";
+                    for (ThreadInfo i : infos) {
+                        message = String.format("%s\n%s", message, i.toString());
+                    }
+                    logger.trace("{}", message);
+                }
+                updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Device Command Queue is Busy");
+            }
+
+        }
+    }
+
+    protected ExecutorService getExecutorService() {
+        return executorService;
+    }
+
+    public String getDeviceId() {
+        return Objects.requireNonNullElse(getThing().getProperties().get(PROP_INFO_DEVICE_ID), "undef");
+    }
+
+    public abstract String getDeviceAlias();
+
+    public abstract String getDeviceUriJsonConfig();
+
+    public abstract void onDeviceRemoved();
+
+    public abstract void onDeviceDisconnected();
+
+    public abstract void updateChannelDynStateDescription() throws LGThinqApiException;
+
+    public abstract LGThinQApiClientService<@NonNull C, @NonNull S> getLgThinQAPIClientService();
+
+    public C getCapabilities() throws LGThinqApiException {
+        if (thinQCapability == null) {
+            thinQCapability = getLgThinQAPIClientService().getCapability(getDeviceId(), getDeviceUriJsonConfig(),
+                    false);
+        }
+        return Objects.requireNonNull(thinQCapability, "Unexpected error. Return of capability shouldn't ever be null");
+    }
+
+    /**
+     * Get the first item value associated to the channel
+     * 
+     * @param channelUID channel
+     * @return value of the first item related to this channel.
+     */
+    @Nullable
+    protected String getItemLinkedValue(ChannelUID channelUID) {
+        Set<Item> items = itemChannelLinkRegistry.getLinkedItems(channelUID);
+        if (!items.isEmpty()) {
+            for (Item i : items) {
+                return i.getState().toString();
+            }
+        }
+        return null;
+    }
+
+    protected abstract Logger getLogger();
+
+    @Override
+    public void initialize() {
+        getLogger().debug("Initializing Thinq thing.");
+
+        Bridge bridge = getBridge();
+        if (bridge != null) {
+            this.account = (LGThinQBridgeHandler) bridge.getHandler();
```

Comment:
Prevents double casting.
```suggestion
        final Bridge bridge = getBridge();
        if (bridge != null && bridge.getHandler() instanceof LGThinQBridgeHandler bridgeHandler) {
            this.account = bridgeHandler;
```